src/test/groovy/graphql/execution/instrumentation/dataloader/DataLoaderHangingTest.groovy

```diff
@@ -154,13 +155,14 @@ class DataLoaderHangingTest extends Specification {
 
         when:
         def graphql = GraphQL.newGraphQL(schema)
-                .instrumentation(new DataLoaderDispatcherInstrumentation(dataLoaderRegistry))
+                .instrumentation(new DataLoaderDispatcherInstrumentation())
                 .build()
 
         then: "execution shouldn't hang"
         List<CompletableFuture<ExecutionResult>> futures = []
         for (int i = 0; i < NUM_OF_REPS; i++) {
-            def result = graphql.executeAsync(ExecutionInput.newExecutionInput()
+            def result = graphql.executeAsync(newExecutionInput()
+                    .dataLoaderRegistry(dataLoaderRegistry)
```

Comment:
it should always be a new DLR, right?

Reply:
yes - it should always be a new DLR - this pattern is to try and encourage that